Modules/posixmodule.c

```diff
@@ -7253,12 +7227,44 @@ os_symlink_impl(PyObject *module, path_t *src, path_t *dst,
 #ifdef MS_WINDOWS
 
     Py_BEGIN_ALLOW_THREADS
-    /* if src is a directory, ensure target_is_directory==1 */
-    target_is_directory |= _check_dirW(src->wide, dst->wide);
-    result = Py_CreateSymbolicLinkW(dst->wide, src->wide,
-                                    target_is_directory);
+    /* if src is a directory, ensure flags==1 (target_is_directory bit) */
+    flags = target_is_directory | _check_dirW(src->wide, dst->wide);
```

Comment:
Wouldn't it be better to merge this region with the next one to have less of them?